api4/post.go

```diff
@@ -941,6 +944,76 @@ func unpinPost(c *Context, w http.ResponseWriter, _ *http.Request) {
 	saveIsPinnedPost(c, w, false)
 }
 
+func acknowledgePost(c *Context, w http.ResponseWriter, r *http.Request) {
+	data := model.MapFromJSON(r.Body)
+	c.RequirePostId()
+	if c.Err != nil {
+		return
+	}
+
+	if !c.App.SessionHasPermissionToUser(*c.AppContext.Session(), data["user_id"]) {
+		c.SetPermissionError(model.PermissionEditOtherUsers)
+		return
+	}
+
+	if !c.App.SessionHasPermissionToChannelByPost(*c.AppContext.Session(), c.Params.PostId, model.PermissionReadChannel) {
+		c.SetPermissionError(model.PermissionReadChannel)
+		return
+	}
+
+	_, appErr := c.App.GetSinglePost(c.Params.PostId, false)
+	if appErr != nil {
+		c.Err = appErr
+		return
+	}
+
+	acknowledgement, appErr := c.App.SaveAcknowledgementForPost(c.AppContext, data["user_id"], c.Params.PostId)
+	if appErr != nil {
+		c.Err = appErr
+		return
+	}
+
+	js, err := json.Marshal(acknowledgement)
+	if err != nil {
+		c.Err = model.NewAppError("acknowledgePost", "api.marshal_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		return
+	}
+
+	w.Write(js)
+}
+
+func unacknowledgePost(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequirePostId()
+	if c.Err != nil {
+		return
+	}
+	userId := r.URL.Query().Get("user_id")
```

Comment:
If we're getting "user_id" from the params, we should probably do a `RequireUserId()` on line 986.

I'm wondering if this is inconsistent with the `acknowledgePost` app method. There we get the "userID" from the request body, but here we get it from the query parameters.

Reply:
@ashishbhate, I was about to change that atm,

Would the most "restful" way be:
 
POST /ack (REQUEST {body: user_id})
DELETE /ack/{user_id}